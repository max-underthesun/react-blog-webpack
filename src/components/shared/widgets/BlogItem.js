import React, { DOM, PropTypes } from 'react';

import { Container, Header } from 'semantic-ui-react';

import TextBox from 'components/shared/elements/TextBox';
import LikeContainer from 'components/containers/LikeContainer';
import MetaData from 'components/shared/elements/Meta';
import Image from 'components/shared/elements/Image';
import Link from 'components/shared/elements/Link';
import { postsPath } from 'helpers/routes';

const BlogItem = ({ id, title, image, text, meta }) => (
  React.createElement(
    Container,
    { className: 'blog-item-container', text: true },
    DOM.div(
      { className: 'blog-item', key: 'mainBlock' },
      React.createElement(
        Header,
        { as: 'h2', className: 'blog-item-header'},
        React.createElement(
          Link,
          { to: postsPath(id)},
          title
        )
      ),
      React.createElement(MetaData, meta),
      React.createElement(Image, image),
      React.createElement(TextBox, text),
      React.createElement(LikeContainer, { meta, id })
    )
  )
);

BlogItem.propTypes = {
  id: PropTypes.number,
  title: PropTypes.string.isRequired,
  image: PropTypes.shape(Image.propTypes).isRequired,
  text: PropTypes.shape(TextBox.propTypes).isRequired,
  meta: PropTypes.shape(MetaData.propTypes).isRequired
};

BlogItem.defaultProps = {
  id: null,
  title: '',
  like: null,
  image: {},
  text: {},
  meta: {}
};

export default BlogItem;
