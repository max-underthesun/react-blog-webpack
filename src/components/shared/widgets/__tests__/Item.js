import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import createStore from 'store';

import Link from 'components/shared/elements/Link';
import BlogItem from '../BlogItem';

describe('BlogItem', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    const store = createStore({});
    // ReactDOM.render(<BlogItem />, div);
    ReactDOM.render(<Provider store={store}><BlogItem /></Provider>, div);
  });

  describe('render', () => {
    it('should render the title', () => {
      const itemProps = { title: 'Hello, World!', id: 5 };

      // const item = shallow(<BlogItem item={itemProps} />);
      const item = shallow(
        React.createElement(BlogItem, itemProps)
      );
      const header = <Link to="/posts/5">Hello, World!</Link>;

      expect(item.contains(header)).toEqual(true);
    });

    it('should render usual item', () => {
      const itemProps = {
        title: 'Hello, World!',
        id: 1,
        image: {
          alt: 'Hello, World',
          src: 'dist/images/img1-lg.jpg'
        }
      };

      const item = shallow(<BlogItem item={itemProps} />);

      expect(item).toMatchSnapshot();
    });
  });
});
