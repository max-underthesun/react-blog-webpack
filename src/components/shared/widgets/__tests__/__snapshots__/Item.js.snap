// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlogItem render should render usual item 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Container
    className="blog-item-container"
    text={true}
>
    <div
        className="blog-item"
    >
        <Header
            as="h2"
            className="blog-item-header"
        >
            <Link
                onlyActiveOnIndex={false}
                style={Object {}}
                to="/posts/null"
            />
        </Header>
        <MetaData
            author="** anonimus **"
        />
        <Image
            height="200px"
            src="/images/no_image.png"
            width="250px"
        />
        <TextBox
            post="** empty entry **"
        />
        <Connect(Like)
            id={null}
            meta={Object {}}
        />
    </div>
</Container>,
  "nodes": Array [
    <Container
      className="blog-item-container"
      text={true}
>
      <div
            className="blog-item"
      >
            <Header
                  as="h2"
                  className="blog-item-header"
            >
                  <Link
                        onlyActiveOnIndex={false}
                        style={Object {}}
                        to="/posts/null"
                  />
            </Header>
            <MetaData
                  author="** anonimus **"
            />
            <Image
                  height="200px"
                  src="/images/no_image.png"
                  width="250px"
            />
            <TextBox
                  post="** empty entry **"
            />
            <Connect(Like)
                  id={null}
                  meta={Object {}}
            />
      </div>
</Container>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <BlogItem
        id={null}
        image={Object {}}
        item={
                Object {
                        "id": 1,
                        "image": Object {
                          "alt": "Hello, World",
                          "src": "dist/images/img1-lg.jpg",
                        },
                        "title": "Hello, World!",
                      }
        }
        like={null}
        meta={Object {}}
        text={Object {}}
        title=""
/>,
      "_debugID": 40,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "id": null,
          "image": Object {},
          "item": Object {
            "id": 1,
            "image": Object {
              "alt": "Hello, World",
              "src": "dist/images/img1-lg.jpg",
            },
            "title": "Hello, World!",
          },
          "like": null,
          "meta": Object {},
          "text": Object {},
          "title": "",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Container
          className="blog-item-container"
          text={true}
>
          <div
                    className="blog-item"
          >
                    <Header
                              as="h2"
                              className="blog-item-header"
                    >
                              <Link
                                        onlyActiveOnIndex={false}
                                        style={Object {}}
                                        to="/posts/null"
                              />
                    </Header>
                    <MetaData
                              author="** anonimus **"
                    />
                    <Image
                              height="200px"
                              src="/images/no_image.png"
                              width="250px"
                    />
                    <TextBox
                              post="** empty entry **"
                    />
                    <Connect(Like)
                              id={null}
                              meta={Object {}}
                    />
          </div>
</Container>,
        "_debugID": 41,
        "_renderedOutput": <Container
          className="blog-item-container"
          text={true}
>
          <div
                    className="blog-item"
          >
                    <Header
                              as="h2"
                              className="blog-item-header"
                    >
                              <Link
                                        onlyActiveOnIndex={false}
                                        style={Object {}}
                                        to="/posts/null"
                              />
                    </Header>
                    <MetaData
                              author="** anonimus **"
                    />
                    <Image
                              height="200px"
                              src="/images/no_image.png"
                              width="250px"
                    />
                    <TextBox
                              post="** empty entry **"
                    />
                    <Connect(Like)
                              id={null}
                              meta={Object {}}
                    />
          </div>
</Container>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <BlogItem
    id={null}
    image={Object {}}
    item={
        Object {
            "id": 1,
            "image": Object {
              "alt": "Hello, World",
              "src": "dist/images/img1-lg.jpg",
            },
            "title": "Hello, World!",
          }
    }
    like={null}
    meta={Object {}}
    text={Object {}}
    title=""
/>,
}
`;
